library/core/src/iter/adapters/zip.rs

```diff
@@ -253,6 +407,40 @@ where
     }
 }
 
+#[inline]
+fn adjust_back_trusted_random_access<
+    A: TrustedRandomAccess + DoubleEndedIterator + ExactSizeIterator,
+    B: TrustedRandomAccess + DoubleEndedIterator + ExactSizeIterator,
+>(
+    zipped: &mut Zip<A, B>,
+) {
+    if A::MAY_HAVE_SIDE_EFFECT || B::MAY_HAVE_SIDE_EFFECT {
+        let sz_a = zipped.a.size();
+        let sz_b = zipped.b.size();
+        // Adjust a, b to equal length, make sure that only the first call
+        // of `next_back` does this, otherwise we will break the restriction
+        // on calls to `zipped.next_back()` after calling `get_unchecked()`.
```

Comment:
Since this has been extracted to a function this comment can be made a doc comment for that function.